backend/src/main/java/com/my/shope/backend/product/ProductService.java

```diff
@@ -103,10 +107,10 @@ public List<Product> getShoppingCart() {
     }
 
 
-    public void removeFromShoppingCart(String productId) {
+    public void removeFromShoppingCartOrOrderedByExecuted(boolean isExecuted, String productId) {
```

Comment:
Ich würde an deiner Stelle hier kein Flag `isExecuted` reinpacken, sondern zwei Methoden machen `removeFromShoppingCart` und `removeFromWhatever`. 
Das machts dann wesentlich verständlicher, wo was gelöscht wird, als mit dem `isExecuted`-Flag.

Den Teil, der für beide gleich ist, würde ich in eine 3. Methode umlagern und diese 3. Methode dann aus den ersten beiden aufrufen.

Reply:
ja das macht sinn aber ich weiss nicht genau wie ich das machen soll